app/src/main/java/com/valhalla/thor/data/provider/ExtensionOpsProvider.kt

```
@@ -0,0 +1,105 @@
+                    // (setAppSuspended → RootMain reflection) is broken on release by R8's class-init
+                    // optimization, and suspend adds nothing for cluster automation. So we ignore the
+                    // user's global Freeze/Suspend preference here and always disable/enable. The
+                    // extension's config UI states that Suspend mode doesn't apply to clusters.
+                    val effective = if (op == Op.TOGGLE) {
+                        if (anyFrozen(ctx.packageManager, targets)) Op.UNFREEZE else Op.FREEZE
+                    } else op
+                    targets.count { pkg ->
+                        when (effective) {
+                            Op.FREEZE -> manageAppUseCase.setAppDisabled(pkg, true)
+                            Op.UNFREEZE -> manageAppUseCase.setAppDisabled(pkg, false)
+                            Op.TOGGLE -> Result.failure(IllegalStateException()) // resolved above
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)